### PR TITLE
Update to Qt 5.6.1

### PR DESCRIPTION
**Pull request by SamGondelman**
- QtWebKit is replaced with QtWebEngine, allowing us to upgrade to Qt 5.6
- remove old WebWindow type (only used in our script-archive) as it was already deprecated and relied on QtWebKit.  Everyone should have already been using OverlayWebWindows anyways.
- enabled automatic URL redirects everywhere (5.6 feature)
- fix some small QML issues
- remove a few now-unneeded Qt hacks

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4783bc05554455519c10e7c754a64473d2f4f58e available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-4783bc0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-4783bc0.dmg

**Comment by SamGondelman** [comment]
- jenkins will try to build this with 5,5 and fail.  Ignore jenkins, he is dumb.
- Links:
     - Windows: http://highfidelity.cachefly.net/HighFidelity-Beta-PR.exe
     - Mac (broken currently, working out some bundling issues): http://highfidelity.cachefly.net/HighFidelity-Beta-PR.dmg

**Comment by SamGondelman** [comment]
Test plan:
- smoke test
- make sure web content (entities, overlays, qml browser) still works
- make sure as many qml windows as you can find still work and don't log any errors
- would be good to test on different graphics cards (particularly nvidia, amd)
- bonus: url redirects should work anywhere you can put a link (I tested this using bit.ly links to existing scripts) (try run script from url, import model, etc)

File: interface/resources/qml/controls-uit/WebView.qml [hunk 1]
```diff
@@ -64,8 +64,4 @@ WebEngineView {
             var newWindow = component.createObject(desktop);
             request.openIn(newWindow.webView)
     }
-
-    // This breaks the webchannel used for passing messages.  Fixed in Qt 5.6
-    // See https://bugreports.qt.io/browse/QTBUG-49521
-    //profile: desktop.browserProfile
```
Comment (jherico) [inline interface/resources/qml/controls-uit/WebView.qml]: why did you remove this?  I get that the bug is fixed, but that should mean that we _can_ set the default profile.

File: interface/resources/qml/controls/WebView.qml [hunk 2]
```diff
@@ -59,8 +59,4 @@ WebEngineView {
             var newWindow = component.createObject(desktop);
             request.openIn(newWindow.webView)
     }
-
-    // This breaks the webchannel used for passing messages.  Fixed in Qt 5.6
-    // See https://bugreports.qt.io/browse/QTBUG-49521
-    //profile: desktop.browserProfile
```
Comment (jherico) [inline interface/resources/qml/controls/WebView.qml]: same

File: interface/src/scripting/WebWindowClass.cpp [hunk 3]
```diff
@@ -69,8 +66,6 @@ WebWindowClass::WebWindowClass(const QString& title, const QString& url, int wid
     }
 
     connect(this, &WebWindowClass::destroyed, _windowWidget, &QWidget::deleteLater);
-    connect(_webView->page()->mainFrame(), &QWebFrame::javaScriptWindowObjectCleared,
```
Comment (jherico) [inline interface/src/scripting/WebWindowClass.cpp]: so I'm pretty sure this is a critical piece for things like edit.js to work.  I don't immediately see how this functionality was replaced.  Can you explain if it was?

Comment (SamGondelman) [inline interface/src/scripting/WebWindowClass.cpp]: Which functionality, specifically?  I believe I can emulate the previous functionality using a QWebChannel but edit.js seems to work fine and I'm not sure I completely understand what the purpose of the event bridge was...

**Comment by hifi-gustavo** [comment]
Build console output for commit: 970d0eb3990b5d6e5d27922c047668226435d59c available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-970d0eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-970d0eb.dmg

File: interface/resources/qml/controls-uit/WebView.qml (hunk 1 above)
Comment (SamGondelman) [inline interface/resources/qml/controls-uit/WebView.qml]: fixed (above in this file and the other one)

**Comment by hifi-gustavo** [comment]
Build console output for commit: f41c537920809d79ef63516e0ba157918bc8051b available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-f41c537.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-f41c537.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dd240a6e1e5bbbff425477d50acd5effba4c8e28 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-dd240a6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-dd240a6.dmg

**Comment by Nex-Pro** [comment]
Crash on adding avatar attachment ( known issue )

developer -> graphics -> shadows **SUCCESS**
developer ->graphics -> ambient occlusion **SUCCESS**
developer -> render -> lod tools **SUCCESS**
developer -> assets -> ATP asset migration **SUCCESS**
developer -> network -> bandwidth details **SUCCESS**
developer -> audio -> show scope **SUCCESS**
developer -> assets -> ATP asset migration **SUCCESS**

**Comment by OmegaHeron** [comment]
@SamGondelman - Tested this on Ubuntu 16.04.1 using Qt5.6.1-1 from Qt's installer.  I'm able to build interface, assignment-client, domain-server, codecs, ice-server and vhacd-util

All produced binaries work as expected

For linux;
I used QT's installer to place content in /opt/hf/Qt5.6.1-1
Added a reference to new library to dynamic library config and updated its cache (ld.so)
Set var for cmake;
export QT_CMAKE_PREFIX_PATH=/opt/hf/Qt5.6.1/5.6/gcc_64/lib/cmake

All looks good - no unexpected compile glitches

**Comment by OmegaHeron** [comment]
@SamGondelman 

Per request - recompiled Interface with web entities enabled (Linux) - no crash looking at Web entity located in standard home scene.  This is with Ubuntu 16.04.1 + QT5.6.1-1 (from QT Installer) on an Alienware Laptop with GTX660M GPU.  

If there's still concern of web entities crashing Linux builds - might I suggest, rather than blanket disable, define a env var to deal with it.  It may well be some users will continue to crash as it's Linux and there's nothing simple about all the various flavors of and levels of software and how it was built.

webentity

**Comment by SamGondelman** [comment]
Thanks @OmegaHeron.  I think I won't worry about it for this PR but I'll put up a note so we can investigate further.

**Comment by Nex-Pro** [comment]
Testing in Win10

file -> help **SUCCESS**
file -> about interface **SUCCESS**

edit -> running scripts **SUCCESS**
***Dialog not closed if re-selected in menu.***
edit -> running scripts -> from url **SUCCESS**
edit -> running scripts -> from disk **SUCCESS**
edit -> running scripts -> remove all **SUCCESS**
edit -> running scripts -> load defaults **SUCCESS**
edit -> running scripts -> reload all **SUCCESS**
edit -> running scripts -> load from picker **SUCCESS**  ( tragically the cow dropped dead after a short while )
edit -> running scripts -> reveal scripts folder **SUCCESS**
edit -> running scripts -> filter **SUCCESS**
edit -> script editor -> **SUCCESS** ( script editing window opens )
***script editor does not open selected file ( known issue )**
edit -> console **SUCCESS**
edit -> asset browser **SUCCESS**
edit -> asset browser -> add to world **SUCCESS**
edit -> asset browser -> rename **SUCCESS**
edit -> asset browser -> delete **SUCCESS**
edit -> asset browser -> choose file **SUCCESS**
edit -> package model **SUCCESS** ( old style look )
edit -> export entities **SUCCESS**
edit -> import entities **SUCCESS**
edit -> import entities from url **SUCCESS**

avatar attachments -> Crash on adding avatar attachment ( known issue )

view -> mini-mirror **SUCCESS**


navigate -> show address bar **SUCCESS**
navigate -> bookmark location **SUCCESS**
navigate -> set home location **SUCCESS**
navigate -> delete bookmark **SUCCESS**

settings -> general **SUCCESS**
settings -> general -> browse (snapshots) **SUCCESS** ***Snapshots folder setting in this and avatar menu below.***
settings -> general -> browse (Load scripts) **SUCCESS**
settings -> general -> load default scripts **Unsure of default behaviour for this item

settings -> avatar -> **SUCCESS**
settings -> avatar -> browse (appearance) **SUCCESS**
settings -> avatar -> browse (snapshots) **SUCCESS** ***Why do we have two places to set the snapshots folder ?***

settings -> audio -> **SUCCESS** **All checkboxes and value boxes working**

settings -> lod -> **SUCCESS**
developer -> graphics -> shadows **SUCCESS**
developer ->graphics -> ambient occlusion **SUCCESS**
developer -> render -> lod tools **SUCCESS**
developer -> assets -> ATP asset migration **SUCCESS**
developer -> network -> bandwidth details **SUCCESS**
developer -> audio -> show scope **SUCCESS**
developer -> assets -> ATP asset migration **SUCCESS**
developer -> assets -> ATP asset migration **SUCCESS**
developer -> assets -> ATP asset migration **SUCCESS**

**Comment by SamGondelman** [comment]
Ok so at the very least it sounds like nothing changed as a result of this PR.  I'm comfortable calling this a QA Pass *on windows*.  Thanks for all the great comments everyone!

Once the mac issues are taken care of, we'll have someone QA that and then we'll be good to go!

**Comment by OmegaHeron** [comment]
QA Pass on Linux as well.

File: interface/src/ui/DataWebPage.cpp [hunk 4]
```diff
@@ -17,38 +17,24 @@
 
 #include "DataWebPage.h"
```
Comment (jherico) [inline interface/src/ui/DataWebPage.cpp]: If you're removing `WebWindowClass` you can remove `DataWebPage` as well.  Nothing else uses it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 01c44434ef0cde1f0f0c35aa486463b21eae8de1 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-01c4443.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-01c4443.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a6b1df2f37f9863bf7afb7d0984b2a80db0f0a8d available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-a6b1df2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-a6b1df2.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ff466feeedf5722fe1c5106286d219a641dda81e available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-ff466fe.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-ff466fe.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6460e8f9bf206ca30dd51b1c1b155663d045e234 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-6460e8f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-6460e8f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e07d388a7e93b6d8380b84181af15b333456ddc0 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-e07d388.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-e07d388.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 764897fc9f1edd1098442e027379a7b95e48b130 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-764897f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-764897f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3f588b6cc825dfa5232fa21d25483de4b540b7c7 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-3f588b6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-3f588b6.dmg

**Comment by marko8904** [comment]
Mac build issues have finally been solved! Here are the links to the latest build:

OS X: http://highfidelity.cachefly.net/HighFidelity-Beta-PR561.dmg
Windows: http://highfidelity.cachefly.net/HighFidelity-Beta-PR561.exe

Windows has already passed previously so I'm comfortable with just focusing on OS X.

As Sam pointed out earlier, here is the test plan:

- smoke test
- make sure web content (entities, overlays, qml browser) still works
- make sure as many qml windows as you can find still work (try edit, browser, etc) and don't log any errors
- would be good to test on different graphics cards (particularly nvidia, amd)
- bonus: url redirects should work anywhere you can put a link (I tested this using bit.ly links to existing scripts) (try run script from url, import model, etc)

**Comment by Nex-Pro** [comment]
**PR Build:** #561
**Testing Architecture:** Desktop
**CPU:** 4.4 ghz Intel i7
**RAM:** 16 GB
**GPU:** Nvidia gtx770 2gb
**Testing OS:** OSX El Capitan
**HMD:** None
**Hand Controllers:** None

**Test Script employed:** http://rawgit.com/cozza13/QAtesting/master/QAofRC.js

**QA Tester**: Piper.Peppercorn

**localhost 'createTestBlocks.js' Result**: SUCCESS
**localhost ATP Upload + Edit Result**: SUCCESS
**Echo Server Audio**: SUCCESS
**localhost 'harmonicOscillator.js' Result**: FAILURE
**localhost 'reverbTest.js' Result**: SUCCESS
**localhost 'playaPerformanceTest.js' Result**: SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: SUCCESS
**QA Pass**: FAILURE


Observations/Feedback:

Possibly related to this PR:

Unable to manipulate new entities created via the edit toolbar. Able to move dynamic objects as normal.

If while selecting one of the platonic solids etc. in the entity properties pane 
it is possible to leave the drop down open by moving the pointer to another location on screen.
Selecting another primitive leaves the drop down open.
With the dropdown open selecting an fbx model acts normally but subsequently selecting a primitive shows the primitive shape selection drop down as open.

Text on text entities looks terrible ( worse than on Win10 )

Harmonic oscillator failed with the cube appearing stationary inside the disk. Both objects were at the same X,Y,Z location and rotation.
This has been seen before.

Url redirect worked for scripts but failed on each attempt to upload models. Models would load from original links but not shortened bitly links.

----------------------------------------------------

Not related to this PR:

ATP:

When uploading a model in the ATP pane it would be nice if the freshly added entity became highlighted and focused in the asset list.
Anything that can be done to reduce the necessity for users to go looking for the new item (especially in the HMD) would be a great benefit in most cases IMO.

Search facility for ATP is absent, would be very useful.


**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |60 |61 |60
**Present Rate**: |60 |60 |60
**Present New Rate**: |0 |0 |0
**CPU Buffer Count**: |4154 |5502 |4700
**GPU Buffer Count**: |1383 |973 |988
**CPU Memory Usage**: |63160320 |128659456 |87203840
**GPU Memory Usage**: |45264896 |51163136 |38674432
**CPU Texture Count**: |292 |571 |376
**GPU Texture Count**: |460 |1000 |777
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |97098012 |1019868336 |342374304
**GPU Texture Memory Usage**: |1067633928 |2342799465 |1144808576
**GPU Virtual Memory Usage**: |1192023496|1706621901|1269198144
**Triangle Count**: |462889 |888561 |839117
**Triangle Rate**: |30859266 |55535062 |49359823
**API Drawcall Count**: |198 |316 |273
**Drawcall Count**: |253 |359 |1196
**Drawcall Rate**: |16866 |22437 |70352
**State Changes Texture Count**:|217 |335 |402
**Pipeline Count**: |39 |41 |59
**Item Opaques**: |269 |469 |270
**Item Translucents**: |18 |57 |75
**Item Lights**: |19 |78 |32
**Timing Opaques**: |708 |1216 |735
**Timing Translucents**: |57 |79 |616
**Timing Lighting**: |21 |31 |30
**Render Frame**: |1280 |2472 |1892


interface -> about interface SUCCESS

file -> help SUCCESS

edit -> running scripts SUCCESS
Dialog not closed if re-selected in menu.
edit -> running scripts -> from url **SUCCESS*
edit -> running scripts -> from disk **SUCCESS**
edit -> running scripts -> remove all **SUCCESS**
edit -> running scripts -> load defaults **SUCCESS**
edit -> running scripts -> reload all  **SUCCESS*
edit -> running scripts -> load from picker  **SUCCESS**
edit -> running scripts -> reveal scripts folder  **SUCCESS**
edit -> running scripts -> filter  **SUCCESS**
edit -> script editor -> SUCCESS ( script editing window opens )
**script editor does not open selected file ( known issue that has been recently fixed)*
edit -> console  **SUCCESS**
edit -> asset browser  **SUCCESS**
edit -> asset browser -> add to world  **SUCCESS**
edit -> asset browser -> rename  **SUCCESS**
edit -> asset browser -> delete  **SUCCESS**
edit -> asset browser -> choose file  **SUCCESS**
edit -> asset browser -> **refresh button does not seem to do anything.**
edit -> package model  **SUCCESS** ( OSX style )
edit -> export entities  **SUCCESS**
edit -> import entities  **SUCCESS** works well
edit -> import entities from url  **SUCCESS**

avatar attachments ->  **SUCCESS**

view -> mini-mirror  **SUCCESS**

navigate -> show address bar **SUCCESS**S
navigate -> bookmark location  **SUCCESS**
navigate -> set home location  **SUCCESS**
navigate -> delete bookmark  **SUCCESS**

settings -> general  **SUCCESS**
settings -> general -> browse (snapshots)  **SUCCESS** Snapshots folder setting in this and avatar menu below.
settings -> general -> browse (Load scripts)  **SUCCESS**
settings -> general -> load default scripts  **SUCCESS**

settings -> avatar ->  **SUCCESS**
settings -> avatar -> browse (appearance)  **SUCCESS**
settings -> avatar -> browse (snapshots)  **SUCCESS** 

settings -> audio -> Checkboxes are invisible when not active.

settings -> lod ->  **SUCCESS**
developer -> graphics -> shadows  **SUCCESS**
developer ->graphics -> ambient occlusion FAIL see image showing tiled texture spread over everything.
developer -> render -> lod tools  **SUCCESS**
developer -> assets -> ATP asset migration  **SUCCESS**
developer -> network -> bandwidth details  **SUCCESS**
developer -> audio -> show scope  **SUCCESS**
developer -> assets -> ATP asset migration  **SUCCESS** ( dialog works but functionality not tested )





ao issue osx

**Comment by Nex-Pro** [comment]
Ok the AO issue shown above is absent on the current release.

**Comment by marko8904** [comment]
Thanks for the thorough QA test @Nex-Pro! Looks like there is an issue for us to address.

**Comment by Nex-Pro** [comment]
@marko8904 The harmonic oscillator also fails for me in 5244 so it is not specific to this PR.

**Comment by marko8904** [comment]
@Nex-Pro Oh! That's good to know!

**Comment by Nex-Pro** [comment]
@marko8904  The radio button issue is specific to this PR it is fine in 5244

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3fd905424deb24ba2f1ac08474163f3cf8e09322 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-3fd9054.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-3fd9054.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9d057233c6fe09f01ee3f19979feef53080d9b51 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-9d05723.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-9d05723.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9d057233c6fe09f01ee3f19979feef53080d9b51 available at: https://gist.github.com/355188962e624f08344b3768e3a05945<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-9d05723.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8405-9d05723.dmg
